packages/orbit-components/src/Textarea/__tests__/index.test.js

```diff
@@ -1,108 +1,75 @@
+    const spaceAfter = SPACINGS_AFTER.NORMAL;
 
-  const component = shallow(
-    <Textarea
-      size={size}
-      name={name}
-      label={label}
-      value={value}
-      fullHeight={fullHeight}
-      placeholder={placeholder}
-      maxLength={maxLength}
-      help={<div>Something useful.</div>}
-      tabIndex={tabIndex}
-      readOnly
-      onChange={onChange}
-      onFocus={onFocus}
-      onBlur={onBlur}
-      dataTest={dataTest}
-      spaceAfter={spaceAfter}
-    />,
-  );
+    render(
+      <Textarea
+        name={name}
+        label={label}
+        value={value}
+        rows={4}
+        fullHeight={fullHeight}
+        placeholder={placeholder}
+        maxLength={maxLength}
+        help={<div>Something useful.</div>}
+        readOnly
+        onChange={onChange}
+        tabIndex={-1}
+        onFocus={onFocus}
+        onBlur={onBlur}
+        dataTest={dataTest}
+        spaceAfter={spaceAfter}
+      />,
+    );
 
-  const area = component.find("Textarea__StyledTextArea");
-  const field = component.find("Textarea__Field");
+    const textarea = screen.getByRole("textbox");
 
-  it("should contain a label", () => {
-    expect(component.find("FormLabel").render().text()).toBe(label);
-  });
-  it("should contain a textarea", () => {
-    expect(area.exists()).toBe(true);
-  });
-  it("should have passed props", () => {
-    expect(area.prop("size")).toBe(size);
-    expect(field.prop("spaceAfter")).toBe(spaceAfter);
-    expect(area.prop("name")).toBe(name);
-    expect(area.prop("value")).toBe(value);
-    expect(area.prop("placeholder")).toBe(placeholder);
-    expect(area.prop("maxLength")).toBe(maxLength);
-    expect(area.prop("fullHeight")).toBe(fullHeight);
-    expect(area.render().prop("data-test")).toBe(dataTest);
-    expect(area.render().prop("readonly")).toBe(true);
-    expect(area.render().prop("tabindex")).toBe(tabIndex);
-    expect(component.find("Textarea__StyledTextArea").render().prop("attribs").name).toBe(name);
-  });
-  it("should contain FeedBack", () => {
-    expect(component.find(`FormFeedback`).exists()).toBe(true);
-  });
-  it("should execute onChange method", () => {
-    area.simulate("change");
+    expect(screen.getByTestId(dataTest)).toBeInTheDocument();
+    expect(textarea).toHaveAttribute("tabindex", "-1");
+    expect(screen.getByDisplayValue(value)).toBeInTheDocument();
+    expect(screen.getByPlaceholderText(placeholder)).toBeInTheDocument();
+    expect(screen.getByText("Something useful.")).toBeInTheDocument();
+    expect(textarea).toHaveAttribute("maxlength", maxLength.toString());
+    expect(textarea).toHaveAttribute("rows", "4");
+    expect(textarea).toHaveAttribute("name", name);
+    expect(textarea.parentElement).toHaveStyle({ marginBottom: "12px" });
+    expect(textarea).toHaveStyle({ padding: "12px" });
+
+    fireEvent.blur(textarea);
+    expect(onBlur).toHaveBeenCalled();
+    // userEvent type does not fire onChange, not sure why, use fireEvent
```

Comment:
It didn't work because you passed `readOnly` to `TextArea`. 😄 I started thinking it was a bug with `@testing-library/user-event` and `textarea`, but then I finally saw it.

Reply:
ah, great, thanks :)